pkg/config/agent/agent.go

```diff
@@ -84,7 +85,22 @@ func DefaultConfig() Config {
 // config struct with merged configuration from all provided files.
 func ParseConfigFiles(files []string) (Config, error) {
 	c := DefaultConfig()
-	return c, cfgutil.ParseYAML(&c, files...)
+	err := cfgutil.ParseYAML(&c, files...)
+
+	mergeConfig(&c, DefaultConfig())
+	return c, err
+}
+
+func mergeConfig(parsed *Config, def Config) {
```

Comment:
Let's put that cfgutil, and use in ParseYAML directly - we would make a shallow copy of the passed pointer and use that as def.